src/actions/ActionHelpers.ts

```diff
@@ -0,0 +1,664 @@
+    let variableName = name;
+
+    if ((name as Variable).name) {
+        variableName = (name as Variable).name;
+    }
+
+    if (typeof variableName !== 'string') {
+        throw new Error('variable argument should be of type "string" but was ' + typeof variableName);
+    }
+
+    try {
+        return studio.variables.getValue(variableName);
+    } catch (error) {
+        throw new VariableNotFoundError(variableName);
+    }
+}
+
+/**
+ * sets the value of a variable
+ * @param name the variable to update
+ * @param value the new variable value
+ */
+export function setVariableValue(name: string | Variable, value: VariableValue) {
+    let variableName = name;
+
+    if ((name as Variable).name) {
+        variableName = (name as Variable).name;
+    }
+
+    if (typeof variableName !== 'string') {
+        throw new Error('variable should be of type string');
+    }
+
+    studio.variables.setValue(variableName, value);
+}
+
+/**
+ * Copies a variable value from one variable to another
+ * @param fromName variable to copy the value from
+ * @param toName variable to copy the value to
+ */
+export function copyVariableValue(fromName: string | Variable, toName: string | Variable) {
+    setVariableValue(toName, getVariableValue(fromName));
+}
+
+/**
+ * Gets all items from the list variable
+ * @param name name of the variable
+ * @returns 
+ */
+export function getAllOptionsFromList(name: string | Variable): string[]{
+    let variableName = name;
+
+    if ((name as Variable).name) {
+        variableName = (name as Variable).name;
+    }
+
+    if (typeof variableName !== 'string') {
+        throw new Error('variable should be of type string');
+    }
+
+    const list = getVariable(variableName) as ListVariable;
+
+    if (list.type !== 'list') {
+        throw new Error('Expected a list variable but got one of type ' + list.type);
+    }
+
+    return list.items;
+}
+
+/**
+ * Gets the selected item from a list variable
+ * @param variable name of the variable
+ * @returns selected item
+ */
+export function getSelectedFromList(variable: string | Variable){
+    let variableName = variable;
+
+    if ((variable as Variable).name) {
+        variableName = (variable as Variable).name;
+    }
+
+    if (typeof variableName !== 'string') {
+        throw new Error('variable should be of type string');
+    }
+
+    const list = getVariable(variableName) as ListVariable;
+
+    if (list.type !== 'list') {
+        throw new Error('Expected a list variable but got one of type ' + list.type);
+    }
+
+    return list.selected;
+}
+
+/**
+ * Set selected list variable item. The same as setVariableValue but with extra runtime checks.
+ * @param name variable name or variable object
+ * @param item item to select, undefined to deselect, or text based variable are also allowed
+ */
+export function setSelectedFromList(name: string | Variable, item: string | VariableValue) {
```

Comment:
`setSelectedItemFromListVariable`

Reply:
Agree on name change - I literally forgot this was a variable for a few seconds